### ISSUE TITLE
Trailing comment at EOF causes a java.lang.Exception: Unexpected EOF

### ISSUE THREAD

**Issue by belucid** [weavejester/cljfmt#22]
When running `lein cljfmt check src/posthere/pretty_print.clj`, I get an error.

The file that causes the issue is here: https://github.com/path/posthere.io/blob/master/src/posthere/pretty_print.clj

The last line of the file ends in a comment:

```clojure
     (mark-url-encoded-invalid request)))) ; potentially mark it as invalid URL encoded
```

Here is the error:

```
java.lang.Exception: Unexpected EOF. [at line 145, column 89]
 at rewrite_clj.reader$throw_reader.doInvoke (reader.clj:18)
    clojure.lang.RestFn.invoke (RestFn.java:425)
    rewrite_clj.reader$read_while.doInvoke (reader.clj:71)
    clojure.lang.RestFn.invoke (RestFn.java:442)
    rewrite_clj.reader$read_until.invoke (reader.clj:80)
    rewrite_clj.reader$read_include_linebreak.invoke (reader.clj:85)
    rewrite_clj.parser.core$eval2038$fn__2039.invoke (core.clj:88)
    clojure.lang.MultiFn.invoke (MultiFn.java:227)
    rewrite_clj.reader$read_with_meta.invoke (reader.clj:115)
    rewrite_clj.parser.core$parse_next.invoke (core.clj:34)
    rewrite_clj.parser$parse.invoke (parser.clj:13)
    rewrite_clj.parser$parse_all$fn__2092.invoke (parser.clj:18)
    clojure.core$repeatedly$fn__4750.invoke (core.clj:4716)
    clojure.lang.LazySeq.sval (LazySeq.java:40)
    clojure.lang.LazySeq.seq (LazySeq.java:49)
    clojure.lang.RT.seq (RT.java:484)
    clojure.core$seq.invoke (core.clj:133)
    clojure.core$take_while$fn__4274.invoke (core.clj:2636)
    clojure.lang.LazySeq.sval (LazySeq.java:40)
    clojure.lang.LazySeq.seq (LazySeq.java:49)
    clojure.lang.Cons.next (Cons.java:39)
    clojure.lang.RT.next (RT.java:598)
    clojure.core$next.invoke (core.clj:64)
    clojure.core$dorun.invoke (core.clj:2856)
    clojure.core$doall.invoke (core.clj:2871)
    rewrite_clj.parser$parse_all.invoke (parser.clj:18)
    rewrite_clj.parser$parse_string_all.invoke (parser.clj:35)
    cljfmt.core$reformat_string.doInvoke (core.clj:211)
    clojure.lang.RestFn.invoke (RestFn.java:423)
    leiningen.cljfmt$reformat_string.invoke (cljfmt.clj:23)
    leiningen.cljfmt$valid_format_QMARK_.invoke (cljfmt.clj:27)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.AFn.applyTo (AFn.java:144)
    clojure.core$apply.invoke (core.clj:626)
    clojure.core$partial$fn__4228.doInvoke (core.clj:2468)
    clojure.lang.RestFn.invoke (RestFn.java:408)
    clojure.core$complement$fn__4082.invoke (core.clj:1370)
    clojure.core$filter$fn__4264.invoke (core.clj:2605)
    clojure.lang.LazySeq.sval (LazySeq.java:40)
    clojure.lang.LazySeq.seq (LazySeq.java:56)
    clojure.lang.Cons.next (Cons.java:39)
    clojure.lang.RT.next (RT.java:598)
    clojure.core$next.invoke (core.clj:64)
    leiningen.cljfmt$check.doInvoke (cljfmt.clj:60)
    clojure.lang.RestFn.applyTo (RestFn.java:142)
    clojure.core$apply.invoke (core.clj:626)
    leiningen.cljfmt$check.invoke (cljfmt.clj:54)
    clojure.lang.AFn.applyToHelper (AFn.java:154)
    clojure.lang.RestFn.applyTo (RestFn.java:132)
    clojure.core$apply.invoke (core.clj:626)
    leiningen.cljfmt$cljfmt.doInvoke (cljfmt.clj:79)
    clojure.lang.RestFn.invoke (RestFn.java:425)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invoke (core.clj:626)
    leiningen.core.main$partial_task$fn__6094.doInvoke (main.clj:263)
    clojure.lang.RestFn.applyTo (RestFn.java:139)
    clojure.lang.AFunction$1.doInvoke (AFunction.java:29)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invoke (core.clj:626)
    leiningen.core.main$apply_task.invoke (main.clj:313)
    lein_environ.plugin$write_env_to_file.invoke (plugin.clj:11)
    clojure.lang.Var.invoke (Var.java:394)
    clojure.lang.AFn.applyToHelper (AFn.java:165)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invoke (core.clj:626)
    robert.hooke$compose_hooks$fn__11735.doInvoke (hooke.clj:40)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.core$apply.invoke (core.clj:624)
    robert.hooke$run_hooks.invoke (hooke.clj:46)
    robert.hooke$prepare_for_hooks$fn__11740$fn__11741.doInvoke (hooke.clj:54)
    clojure.lang.RestFn.applyTo (RestFn.java:137)
    clojure.lang.AFunction$1.doInvoke (AFunction.java:29)
    clojure.lang.RestFn.invoke (RestFn.java:436)
    leiningen.core.main$resolve_and_apply.invoke (main.clj:319)
    leiningen.core.main$_main$fn__6160.invoke (main.clj:392)
    leiningen.core.main$_main.doInvoke (main.clj:385)
    clojure.lang.RestFn.invoke (RestFn.java:421)
    clojure.lang.Var.invoke (Var.java:383)
    clojure.lang.AFn.applyToHelper (AFn.java:156)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.core$apply.invoke (core.clj:624)
    clojure.main$main_opt.invoke (main.clj:315)
    clojure.main$main.doInvoke (main.clj:420)
    clojure.lang.RestFn.invoke (RestFn.java:457)
    clojure.lang.Var.invoke (Var.java:394)
    clojure.lang.AFn.applyToHelper (AFn.java:165)
    clojure.lang.Var.applyTo (Var.java:700)
    clojure.main.main (main.java:37)
```

**Comment by xsc** [comment]
This bug should have been fixed. Do you maybe have another plugin active that uses `rewrite-clj`, e.g. `lein-ancient` or `lein-plz`? Try sticking the following into your `:plugins` vector to verify that that is the problem:

```clojure
  :plugins [... [rewrite-clj "0.4.11"] ...]
```

**Comment by weavejester** [comment]
It's your use of lein-ancient that's causing the problem:

```
$ lein deps :tree
Possibly confusing dependencies found:
[lein-ancient "0.6.1"] -> [rewrite-clj "0.4.4"]
 overrides
[lein-cljfmt "0.1.5"] -> [cljfmt "0.1.5"] -> [rewrite-clj "0.4.10"]
```

You'll either need to remove that dependency, or force a later version of rewrite-clj by adding it to your plugins, as @xsc describes.

**Comment by xsc** [comment]
I released [lein-ancient 0.6.2](https://github.com/xsc/lein-ancient/releases/tag/clojars-0.6.2) which uses the latest 'rewrite-clj`. Upgrading should solve this problem.

**Comment by belucid** [comment]
That was the problem. New lein-ancient resolved it. Thanks @weavejester and @xsc
